### PR TITLE
libdl linkage fixes

### PR DESCRIPTION
**Pull request by justinboswell**
* Separated linkage on Darwin from Linux/glibc
* Updated builder dependency, added support for .NET 5.0, 6.0
* CI is running tests again, added check-submodules job
* Made use of `aws_split_debug_info` to get the Linux .so down to 3MB 

Depends on:
* https://github.com/awslabs/aws-c-common/pull/819 (replaces https://github.com/awslabs/aws-crt-dotnet/pull/62)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: nuget.config [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <configuration>
     <packageSources>
         <add key="local" value="packages" />
+        <add key="global" value="/home/boswej/.nuget/packages"/>
```
Comment (bretambrose) [inline nuget.config:5]: Probably don't want to add this.

File: tests/tests.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.1</TargetFrameworks>
+    <TargetFrameworks>netcoreapp3.1</TargetFrameworks>
```
Comment (bretambrose) [inline tests/tests.csproj:4]: Does this property support multiple frameworks?

**Review by bretambrose — APPROVED** [review]
Ship it assuming aws-c-common tagged release

File: aws-crt/CRT.cs [hunk 3]
```diff
@@ -297,42 +340,48 @@ public override LibraryHandle LoadLibrary(string name)
...
+        internal static PlatformLoader Loader
         {
             get {
                 if (s_loader != null)
                 {
                     return s_loader;
                 }
 
-                if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```
Comment (ashovlin) [inline aws-crt/CRT.cs:371]: This isn't available for `net35/net45` - we do have a shim in `Aws.Crt.Platform.GetRuntimePlatformOS()`.

(looks like Windows CI is running `dotnet build -f netstandard2.0` - unsure if the images are set up for it, but exercising .NET Framework here too might be nice)

Comment (justinboswell) [reply aws-crt/CRT.cs:371]: Oh, I can detect windows another way, good call.

**Pull request merged**